kvserver: TestRangeStatsInit fails under stress

**Describe the problem**

On cf2b7f81ceeb3b44c4a2398218f7902f901d338d, `TestRangeStatsInit` fails under stress:

```
--- FAIL: TestRangeStatsInit (0.11s)                                                                                                                                                                                    
    test_log_scope.go:161: test logs captured to: /Users/ssd/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/d5a35cf87725f62ad73d3efbbbf6e538/logTestRangeStatsInit3433310875                                          
    test_log_scope.go:79: use -show-logs to present logs inline                                                                                                                                                         
    stats_test.go:55: mvcc stats mismatch {ContainsEstimates:0 LastUpdateNanos:11 IntentAge:9 GCBytesAge:10 LiveBytes:1 LiveCount:5 KeyBytes:2 KeyCount:6 ValBytes:3 ValCount:7 IntentBytes:4 IntentCount:8 SeparatedInt
entCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:0 SysCount:0 AbortSpanBytes:0} != {ContainsEstimates:0 LastUpdateNanos:123 IntentAge:0 GCBytesAge:0 LiveBytes:35637 LiveCount:105 Ke
yBytes:2686 KeyCount:105 ValBytes:32951 ValCount:105 IntentBytes:0 IntentCount:0 SeparatedIntentCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:260 SysCount:6 AbortSpanBytes:0}       
    stats_test.go:57: -- test log scope end --                                                              
FAIL                                                                                                        
I221110 15:03:20.911461 1 (gostd) testmain.go:1476  [-] 1  Test //pkg/kv/kvserver:kvserver_test exited with error code 1
```

It usually fails in < 200 runs.



Jira issue: CRDB-21377

--- a/pkg/kv/kvserver/stats_test.go
+++ b/pkg/kv/kvserver/stats_test.go
@@ -30,6 +30,11 @@ func TestRangeStatsInit(t *testing.T) {
 	stopper := stop.NewStopper()
 	defer stopper.Stop(ctx)
 	tc.Start(ctx, t, stopper)
+
+	// Lock the replica's Raft goroutine. We do this to avoid interference from
+	// any other moving parts on the Replica, whatever they may be.
+	tc.repl.raftMu.Lock()
+	defer tc.repl.raftMu.Unlock()
 	ms := enginepb.MVCCStats{
 		LiveBytes:       1,
 		KeyBytes:        2,

